plugins/source/azure/codegen/recipes/base.go

```diff
@@ -0,0 +1,260 @@
+}
+
+type template struct {
+	source            string
+	destinationSuffix string
+	imports           []string
+}
+
+type resourceDefinition struct {
+	customColumns            codegen.ColumnDefinitions
+	azureStruct              interface{}
+	skipFields               []string
+	includeColumns           string
+	helpers                  []string
+	isRelation               bool
+	listFunction             string
+	listFunctionArgs         []string
+	listFunctionArgsInit     []string
+	listHandler              string
+	getFunction              string
+	getFunctionArgs          []string
+	getFunctionArgsInit      []string
+	mockListResult           string
+	mockHelpers              []string
+	mockListFunctionArgs     []string
+	mockListFunctionArgsInit []string
+	mockFieldsToIgnore       []string
+	mockValueType            string
+	mockDefinitionType       string
+	subServiceOverride       string
+	relations                []string
+}
+
+type byTemplates struct {
+	templates           []template
+	definitions         []resourceDefinition
+	serviceNameOverride string
+}
+
+const (
+	pluginName   = "azure"
+	valueHandler = `if err != nil {
+		return errors.WithStack(err)
+	}
+	if response.Value == nil {
+		return nil
+	}
+	res <- *response.Value
+	`
+)
+
+var (
+	subscriptionIdColumn = codegen.ColumnDefinition{
+		Name:     "subscription_id",
+		Type:     schema.TypeString,
+		Resolver: "client.ResolveAzureSubscription",
+	}
+	defaultSkipFields = []string{"Response"}
+)
+
+func AllResources() []Resource {
+	var resources = []Resource{}
+	resources = append(resources, Authorization()...)
+	resources = append(resources, Batch()...)
+	resources = append(resources, CDN()...)
+	resources = append(resources, Compute()...)
+	resources = append(resources, Container()...)
+	resources = append(resources, CosmosDB()...)
+	resources = append(resources, Datalake()...)
+	resources = append(resources, EventHub()...)
+	resources = append(resources, FrontDoor()...)
+	resources = append(resources, IotHub()...)
+	resources = append(resources, Network()...)
+	resources = append(resources, KeyValue()...)
+	resources = append(resources, Logic()...)
+	resources = append(resources, MariaDB()...)
+	resources = append(resources, Monitor()...)
+	resources = append(resources, MySQL()...)
+	resources = append(resources, PostgresSQL()...)
+	resources = append(resources, Redis()...)
+	resources = append(resources, Resources()...)
+	resources = append(resources, Search()...)
+	resources = append(resources, Security()...)
+	resources = append(resources, ServiceBus()...)
+	resources = append(resources, SQL()...)
+	resources = append(resources, Storage()...)
+	resources = append(resources, StreamAnalytics()...)
+	resources = append(resources, Subscriptions()...)
+	resources = append(resources, Web()...)
+	return resources
+}
+
+func needsSubscriptionId(table *codegen.TableDefinition) bool {
+	for _, column := range table.Columns {
+		if column.Name == "subscription_id" {
```

Comment:
Some resources have `subscription_id` column returned from the API so no need to resolve it